CMakePresets.json

```diff
@@ -29,24 +21,66 @@
     {
       "name": "gcc-debug",
       "displayName": "GCC Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined"
+      }
+    },
+    {
+      "name": "gcc-release",
+      "displayName": "GCC Release Build",
       "inherits": [
         "_root-config",
-        "_debug-base"
+        "_release-base"
       ],
       "cacheVariables": {
-        "CMAKE_CXX_COMPILER": "g++"
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined"
```

Comment:
Do not override the default release `CMAKE_CXX_FLAGS`.
Delete this line.

Reply:
May I ask why?

Reply:
Because this is the `gcc-release` configure preset!

Reply:
Ah I see, nice catch.